src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -376,6 +376,11 @@ public unsafe void SetResponseHeaders()
                 var knownHeaderIndex = HttpApiTypes.HTTP_RESPONSE_HEADER_ID.IndexOfKnownHeader(headerPair.Key);
                 for (var i = 0; i < headerValues.Count; i++)
                 {
+                    if(string.IsNullOrEmpty(headerValues.ToString()))
```

Comment:
There are two null conditions to check for here.
1) if headerValues is empty (it's a struct, it can't be null) - This should be checked before the for loop by looking at the headerValues.Count.
https://github.com/aspnet/AspNetCore/blob/fad6a1867ccb42bcf23c8507218b944e8798a06a/src/Servers/HttpSys/src/RequestProcessing/Response.cs#L477-L480
2) if any entry in headerValues is null. - This you need to check for every `headerValues[i]`.
https://github.com/aspnet/AspNetCore/blob/fad6a1867ccb42bcf23c8507218b944e8798a06a/src/Servers/HttpSys/src/RequestProcessing/Response.cs#L507

Reply:
1. I believe this was already the case, as it would just skip over the loop if the StringValues struct was empty. I'm all for clarity so I added the count check.

2. Ah yes, obvious in hind sight I need to check the actual value we are iterating over in the loop!